### PR TITLE
Rewrite responsive navbar

### PR DESCRIPTION
**Pull request by JulianDietzel**
I created a responsive navbar with the following five routes:
- Tricks: `/tricks`
- Combos: `/combos`
- Glossary: `/glossary`
- Generator (Combo Generator): `/generator`
- More (This is where I'd put the few settings we have, the imprint and a short About section): `/more`

The navbar comes in only a light theme for now as I think that this is what we should design for first and has language support for English, Spanish and German. It is responsive, switching from a bottom navigation to a side bar at the `lg`-breakpoint of Tailwind. I'm going back and forth whether the `xl`-breakpoint should be used. I think the way I did it is the way though let me know if you know any good reasons for using one or the other.

| Mobile | Desktop |
| - | - |
|  | Bildschirmfoto am 2023-11-11 um 18 46 27 |

I also added one npm script which serves the website in a manner so that other devices of the network can access the page. It can be run with `npm run host`.

@bastislack @Dosbodoke  Le me know if you have any feedback on the design.
@WaldemarLehner Can you take a look and make sure I didn't mess up anything with Vue / TailwindCSS.

### CODE REVIEW

**Comment by JulianDietzel** [comment]
After talking to @hell3nacht about it, we slightly changed the highlighting of the router items. I also changed the background to white in order to actually see the navbar with the correct matching background color:

| Mobile | Desktop |
| - | - |
|  |  |

**Comment by WaldemarLehner** [comment]
I havent looked at the code yet... Yet I would still do one Suggestion in regardsvto path naming.

We should stick to a REST-like naming scheme as much as possible.

This would mean getting all combos at /combos, getting a specific combo at /combos/$id and so on.

I think that /generate breaks that convention. I would suggest /combos/new or /combos/create or /combos/new/generate if generation should be wildly different from 'manual' creation .



I would also rename /more to /about, as it gives a clearer description of what this page does.

File: README.md
Comment (WaldemarLehner) [inline README.md:61]: You could also `npm run dev -- --host` to pass the argument to vite.

File: src/components/navbars/NavbarBottom.vue
Comment (WaldemarLehner) [inline src/components/navbars/NavbarBottom.vue]: When I set up the example I made `/i18n/index` have the keys for the Index-Route.
What do you think about storing the Strings under `/i18n/navbars` instead?

File: src/components/navbars/NavbarBottom.vue
Comment (WaldemarLehner) [inline src/components/navbars/NavbarBottom.vue]: nit: translationKey? messageKey?

File: src/layouts/DefaultLayout.vue
Comment (WaldemarLehner) [inline src/layouts/DefaultLayout.vue]: What are the implications of using `visibility: collapse` over `display: none`?

File: src/components/navbars/NavbarSide.vue
Comment (WaldemarLehner) [inline src/components/navbars/NavbarSide.vue]: `/about` would be more fitting

File: src/components/navbars/NavbarSide.vue
Comment (WaldemarLehner) [inline src/components/navbars/NavbarSide.vue]: `/combos/new` or `/combos/new/generate` would be more descriptive and would more closely model REST-Like Structures

**Review by WaldemarLehner — CHANGES_REQUESTED** [review]

File: README.md
Comment (JulianDietzel) [reply README.md:61]: Would that be better? Serious question, are there any benefits or downsides to either having a `host` script or passing the `--host` flag and not having a script?

File: src/components/navbars/NavbarBottom.vue
Comment (JulianDietzel) [reply src/components/navbars/NavbarBottom.vue]: Seems like a good idea. What do you think of the following structure (and is it even a good approach to have multiple .ts files and to have them in separate folders):
```
i18n
    index
        index.ts
        index.de.json
        index.en.json
        ...
    navbar
        navbar.ts
        navbar.de.json
        navbar.en.json
        ...
```

File: src/components/navbars/NavbarBottom.vue
Comment (JulianDietzel) [reply src/components/navbars/NavbarBottom.vue]: Renamed to `translationKey`. Fixed.

File: src/layouts/DefaultLayout.vue
Comment (JulianDietzel) [reply src/layouts/DefaultLayout.vue]: [Relevant docs](https://tailwindcss.com/docs/visibility)

I just used `collapse` as it was the most sensible option out of the visibility options (`visible`, `invisible` and `collapse`) and didn't even consider using `display: none`. Looking at it now, `display: none` might be more suitable as it allows for other components to change their size. I'll try switching over to this.

Comment (JulianDietzel) [reply src/layouts/DefaultLayout.vue]: Fixed by switching to `hidden` class which internally sets `display: none` in order to hide navbars. Used `block` as inverse class.
Kept `collapse` and `visible` for the text on the bottom navbar as this way the icon's size can not change (not a real concern, but I think it is more appropriate to use these to options here).

File: src/components/navbars/NavbarSide.vue
Comment (JulianDietzel) [reply src/components/navbars/NavbarSide.vue]: Agreed. I think `/combos/new` would not be good, as to me that route seems more appropriate for the page where a user can create (not generate) a custom combo. 
Would `/combos/new/generate` be better in any way than `/combos/generate`? Otherwise I'd go with the `/combos/generate` option.

File: src/components/navbars/NavbarSide.vue
Comment (JulianDietzel) [reply src/components/navbars/NavbarSide.vue]: That is what I called it at first. Then I changed it, as there might be settings under that page as well. Though I think you are still correct and at least the route should be called `/about`. I'll still leave the name of the button ("More") the same though.

Comment (JulianDietzel) [reply src/components/navbars/NavbarSide.vue]: Changed route from `/more` to `/about`, left button name the same ("More").

File: README.md
Comment (WaldemarLehner) [reply README.md:61]: oh no. Keep the `host`script :) 
I just figured the script was added you didn't know about this `-- --$params` notation. If you feel like it would be used a lot, feel free to keep it.

I reckon its useful for checking on Mobile Devices.

Feel free to resolve conversation.

File: src/components/navbars/NavbarBottom.vue
Comment (WaldemarLehner) [reply src/components/navbars/NavbarBottom.vue]: yup. That looks good!

File: src/components/navbars/NavbarSide.vue
Comment (WaldemarLehner) [reply src/components/navbars/NavbarSide.vue]: The question is… in what way do we wish to implement generation.

If it is logically a subpage of creating a new combo, e.g. you click on "Create new" and then click a Dice-Button or something like that, I would vouch for `/combos/new/generate`. 

If `/combos/new` and `/combos/generate` can be seen as fundamentally different routes, `/combos/generate` would work. 

To me it seems that generation would be fundamentally tied to creation. But it really depends on the impl.


I was thinking about a Settings-Button on `/combos/new` which would create a popup where you can select your params and a Dice-Button which would reroll for a new Combo.

The "result" would be placed into the List of Tricks which you could also get by manually adding tricks to the combo. 
In that sense, generation would basically be pre-filling the contents on the `/combos/new` page and you wouldnt even need a `/combos/new/generate` 


So yeah… depends on the impl :P

File: README.md
Comment (JulianDietzel) [reply README.md:61]: Sweet, thanks for the explanation. Seeing the webpage on Mobile was the exact reason I created the command.

File: src/components/navbars/NavbarBottom.vue
Comment (JulianDietzel) [reply src/components/navbars/NavbarBottom.vue]: Implemented it this exact way. Grouped the `index`, `navbar` and `notFound` translations into new separate folders.

File: src/components/navbars/NavbarSide.vue
Comment (JulianDietzel) [reply src/components/navbars/NavbarSide.vue]: From all that I have seen and heard so far, the Combo Generator will be a very separate feature with its own tab in the navbar. The process currently (and most probably going forwards) is, that you generate a combo and in case you like it a lot you can save it, which is done by then opening the menu for manually creating a combo and automatically filling in the tricks from the generator.

Therefore, if I read your answer correctly, using `/combos/generate` should be alright shouldn't it?

Comment (WaldemarLehner) [reply src/components/navbars/NavbarSide.vue]: Yup. combos/generate works for me for now. 
I would like some input from the rest of the team about how they envision `generate` to work, but that's out of scope for now :) 

feel free to resolve

**Comment by JulianDietzel** [comment]
Currently blocked by unresolved issue #305 (missing component library)

**Comment by JulianDietzel** [comment]
#305 has been resolved by PR #311 . Now further work is blocked by the unmerged PR #316. The change is needed in order to be able to use the primary color with different brightness levels. Once that is merged there should be no further issues and I can finish the navbar.

**Comment by JulianDietzel** [comment]
Replaced by PR #318.

**Pull request closed without merging**
